test/e2e/networking.go

```diff
@@ -121,8 +121,21 @@ var _ = Describe("Networking", func() {
 		filterNodes(nodes, func(node api.Node) bool {
 			return isNodeReadySetAsExpected(&node, true)
 		})
+
 		if len(nodes.Items) < 2 {
-			Failf("Less than two nodes were found Ready.")
+			// in general, the test requires two nodes. But for local development, often a one node cluster
+			// is created, for simplicity and speed. (see issue #10012). We permit one-node test
+			// only in some cases
+			if providerIs("gce", "gke") {
```

Comment:
I would prefer if you flipped this condition around; only the local cluster thingy should be allowed to pass this with a single node.